### ISSUE TITLE
Investigate fabtests failures

### ISSUE THREAD

**Issue by sungeunchoi** [ofi-cray/libfabric-cray#900]
Some of these should be exiting with ENODATA:
- fi_size_left_test
- fi_dgram_waitset
- fi_poll
- fi_rdm_tagged_peek (started failing 7/13, possibly due to #859)
- fi_rdm_shared_av
- fi_rdm_cntr_pingpong

**Comment by sungeunchoi** [comment]
fi_size_left_test:

```
  Running rx_size_left...fi_size_left_test: ./include/fi_atom.h:71: atomic_inc: Assertion `atomic->is_initialized' failed.
```

fi_dgram_waitset:

```
 server_stdout: |
    fi_wait(): simple/dgram_waitset.c:102, ret=-38 (Function not implemented)
    Posting a send...
    libfabric:gni:eq:gnix_wait_close():272<warn> [22828:1] resources still connected to wait set.
    [error] fabtests:common/shared.c:822: fi_close (-16) fid 16
  client_stdout: |
    Posting a send...
    fi_wait(): simple/dgram_waitset.c:102, ret=-38 (Function not implemented)
    libfabric:gni:eq:gnix_wait_close():272<warn> [14694:1] resources still connected to wait set.
    [error] fabtests:common/shared.c:822: fi_close (-16) fid 16
```

fi_poll:

```
  server_stdout: |
    fi_poll_open(): simple/poll.c:60, ret=-38 (Function not implemented)
  client_stdout: |
    fi_poll_open(): simple/poll.c:60, ret=-38 (Function not implemented)
```

fi_rdm_tagged_peek:

```
  server_stdout: |
    Killed by signal 15.
  client_stdout: |
    Searching for a bad msg
    FI_PEEK(): simple/rdm_tagged_peek.c:58, ret=-22 (Invalid argument)
    FI_PEEK(): simple/rdm_tagged_peek.c:84, ret=-22 (Invalid argument)

```

fi_rdm_shared_av:

```
  server_stdout: |
    Killed by signal 15.
  client_stdout: |
```

fi_rdm_cntr_pingpong:

```
  server_stdout: |
    receive(): common/shared.c:1170, ret=-263 (Missing or unavailable completion queue)
  client_stdout: |
    receive(): common/shared.c:1170, ret=-263 (Missing or unavailable completion queue)
```

**Comment by sungeunchoi** [comment]
- fi_size_left_test: Proposed fix in https://github.com/ofiwg/fabtests/pull/543
- fi_rdm_tagged_peek: Proposed fix in #901 
- Others are expected (i.e., APIs not implemented)

**Comment by sungeunchoi** [comment]
The fi_size_left_test issue is still open.

**Comment by sungeunchoi** [comment]
Update:
- fi_size_left_test: Test fixed upstream, we should address it with #902 
- fi_rdm_tagged_peek: Fixed
- Others expected
